src/vm/gcenv.ee.cpp

```diff
@@ -1551,3 +1551,45 @@ void GCToEEInterface::VerifySyncTableEntry()
     SyncBlockCache::GetSyncBlockCache()->VerifySyncTableEntry();
 #endif // VERIFY_HEAP
 }
+
+#if defined(__linux__)
+void GCToEEInterface::UpdateGCEventStatus()
+{
+    LIMITED_METHOD_CONTRACT;
+    // LTTng does not have a notion of enabling events via "keyword"/"level" but we have to 
+    // somehow implement a similar behavior to it. 
```

Comment:
FWIW, I don't think we have to implement keywords/levels but it might be convenient to because it induces less code churn (and I'm fine with this just how it is). Below is just understanding an alternative if it is useful for the future:

I only identified one spot in the GC which checks whether a keyword is enabled prior to triggering the event, line 31989 of [gc.cpp](https://raw.githubusercontent.com/dotnet/coreclr/master/src/gc/gc.cpp). Given that site is checking for GCEventKeyword_GCHeapSurvivalAndMovement and there is only one event with that keyword, you could imagine changing that site to explicitly test the GCGenerationRange event rather than the keyword. Then all the EVENT_ENABLED(name) macros could be redefined to check a specific bit per event rather than calculating the bit from synthesized keyword and level. The last part would be changing this to capture 1 bit per event for all events rather than just selected events that we are using as canonical indicators.

Reply:
are you proposing this for all events or just this one particular event? I understand if it's the latter but I don't understand the former 'cause if we did individual bit for each event we'd have to get that info for every event we possibly want to fire in every GC? I thought you said that was not easy to get.

Reply:
> are you proposing this for all events or just this one particular event?

All events (I am not proposing we do it, just commenting that alternatives do exist)

> I thought you said that was not easy to get.

Getting the bit for an event means calling EventXplatEnabledXXX(), the code below shows that in action for seven events. The alternative I described would gather bits for all ~40 events that the GC uses. Code-wise its pretty formulaic and could be driven with defines in the known event list rather than hardcoding. On the perf-side obviously more queries is slower than fewer queries though I'd guess each one is < 10 cycles so I suspect you would have a hard time measuring a difference.